x-pack/plugins/apm/server/routes/metrics/serverless/helper.test.ts

```diff
@@ -38,3 +42,85 @@ describe('calcMemoryUsedRate', () => {
     expect(calcMemoryUsedRate({ memoryFree: 50, memoryTotal: 100 })).toBe(0.5);
   });
 });
+
+const AWS_LAMBDA_PRICE_FACTOR = {
+  x86_64: 0.0000166667,
+  arm: 0.0000133334,
+};
+
+describe('calcEstimatedCost', () => {
+  it('returns 0 when price factor is not defined', () => {
+    expect(
+      calcEstimatedCost({
+        totalMemory: 1,
+        billedDuration: 1,
+        transactionThroughput: 1,
+        architecture: 'arm',
+      })
+    ).toEqual(0);
+  });
+
+  it('returns 0 when architecture is not defined', () => {
+    expect(
+      calcEstimatedCost({
+        totalMemory: 1,
+        billedDuration: 1,
+        transactionThroughput: 1,
+        awsLambdaPriceFactor: AWS_LAMBDA_PRICE_FACTOR,
+      })
+    ).toEqual(0);
+  });
+
+  it('returns 0 when compute usage is not defined', () => {
+    expect(
+      calcEstimatedCost({
+        transactionThroughput: 1,
+        awsLambdaPriceFactor: AWS_LAMBDA_PRICE_FACTOR,
+        architecture: 'arm',
+      })
+    ).toEqual(0);
+  });
+
+  it('returns 0 when request cost per million is not defined', () => {
+    expect(
+      calcEstimatedCost({
+        totalMemory: 1,
+        billedDuration: 1,
+        transactionThroughput: 1,
+        awsLambdaPriceFactor: AWS_LAMBDA_PRICE_FACTOR,
+        architecture: 'arm',
+      })
+    ).toEqual(0);
+  });
+
+  describe('x86_64 architecture', () => {
+    const architecture = 'x86_64';
+    it('returns correct cost when usage is less than 6 b gb-sec', () => {
+      expect(
+        calcEstimatedCost({
```

Comment:
I think it'd be useful to have several invocations here, it would make it more clear from just reading the test how different factors influence the final result.